src/gravel/controllers/orch/system.py

```diff
@@ -32,9 +41,23 @@ def set_address(addr: str) -> bool:
     try:
         subprocess.check_output(restart_cmd, stderr=subprocess.PIPE)
     except subprocess.CalledProcessError as e:
-        raise ChronyRestartError("Unable to restart chronyd.service") from e
+        raise SystemCtlError("Failed to restart unit 'chronyd.service'.") from e
     return True
 
 
-class ChronyRestartError(Exception):
-    pass
+def set_hostname(name: str) -> bool:
```

Comment:
This needs to be gated with a check on whether the node has been inited or not. That should be done with a call to the nodemgr. From this path, you can obtain that from the API's endpoint handler's `Request` field and pass it on to this function (or, better yet, check it directly in the api endpoint handler).

Reply:
Please check if the test and exception meets your expectation.